### PR TITLE
Adding metadata instance tags and deleting invalid metadata option

### PR DESCRIPTION
**Pull request by jamengual**
## what
* Adding `instance_metadata_tags`
* Deleting invalid invalid metadata ipv6 option

## why
* To allow the instance to grab its own tags

## references
- https://registry.terraform.io/providers/hashicorp/aws/4.37.0/docs/resources/instance#instance_metadata_tags
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#http_protocol_ipv6
- Previous PR https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/pull/79

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -105,7 +105,7 @@ resource "aws_launch_template" "default" {
     http_endpoint               = (var.metadata_http_endpoint_enabled) ? "enabled" : "disabled"
     http_put_response_hop_limit = var.metadata_http_put_response_hop_limit
     http_tokens                 = (var.metadata_http_tokens_required) ? "required" : "optional"
-    http_protocol_ipv6          = (var.metadata_http_protocol_ipv6_enabled) ? "enabled" : "disabled"
+    instance_metadata_tags      = (var.metadata_instance_metadata_tags_enabled) ? "enabled" : "disabled"
```
Comment (nitrocode) [inline main.tf:108]: Why remove this line?

```suggestion
    http_protocol_ipv6          = (var.metadata_http_protocol_ipv6_enabled) ? "enabled" : "disabled"
    instance_metadata_tags      = (var.metadata_instance_metadata_tags_enabled) ? "enabled" : "disabled"
```

File: variables.tf [hunk 2]
```diff
@@ -455,14 +455,14 @@ variable "metadata_http_put_response_hop_limit" {
 
 variable "metadata_http_tokens_required" {
   type        = bool
-  default     = true
+  default     = false
```
Comment (nitrocode) [inline variables.tf:458]: Why change this default ?

```suggestion
  default     = true
```

File: variables.tf [hunk 3]
```diff
@@ -455,14 +455,14 @@ variable "metadata_http_put_response_hop_limit" {
...
   description = "Set true to require IMDS session tokens, disabling Instance Metadata Service Version 1."
 }
 
-variable "metadata_http_protocol_ipv6_enabled" {
+variable "metadata_instance_metadata_tags_enabled" {
```
Comment (nitrocode) [inline variables.tf:462]: Why rename this variable ?

File: main.tf (hunk 1 above)
Comment (nitrocode) [reply main.tf:108]: This seems wrong. You're removing http_protocol_ipv6 from the aws_launch_template resource and instance_metadata_tags is also an option here

Comment (jamengual) [reply main.tf:108]: is not a valid metadata option, I checked the provider back to version 3, it never existed

**Pull request closed without merging**